packages/e2e-tests/src/utilities.ts

```diff
@@ -0,0 +1,84 @@
+import path from 'node:path';
+import process from 'node:process';
+import * as core from '@actions/core';
+import child_process, { type SpawnSyncReturns, type SpawnSyncOptionsWithStringEncoding } from 'node:child_process';
+
+export function command(binary: string, group?: string): boolean {
+	const bin = 'command';
+	const args = ['-v', binary] as const;
+	const options: SpawnSyncOptionsWithStringEncoding = { stdio: 'pipe', encoding: 'utf-8', shell: true };
+	const command = child_process.spawnSync(bin, args, options);
+	if (command.status !== 0) {
+		if (group) {
+			core.startGroup(group);
+		}
+		core.error(`Did not find installed binary for '${binary}'!`);
+		if (group) {
+			core.endGroup();
+		}
+		return false;
+	}
+	return true;
+}
+
+export function logSpawnSyncError({
+	command,
+	group,
+	message,
+}: {
+	group?: string;
+	message?: string;
+	command: SpawnSyncReturns<string | Buffer>;
+}): void {
+	if (group) {
+		core.startGroup(group);
+	}
+	if (message) {
+		core.error(message);
+	}
+	core.error('Stderr: ' + command.stderr);
+	core.error('Signal: ' + command.signal);
+	core.error('Status: ' + command.status);
+	if (command.error) {
+		core.error('Error: ' + command.error.message);
+	}
+	if (group) {
+		core.endGroup();
+	}
+}
+
+type LogType = keyof Pick<typeof core, 'notice' | 'info' | 'warning' | 'error' | 'debug'>;
+
+type LogOptions = { type?: LogType; group?: string };
+
+export function log(message: string | string[], options: LogOptions = { type: 'error' }): void {
+	const type = options.type ?? 'error';
+	const group = options.group;
```

Comment:
no default defined, and not required, so maybe something like:

```js
const group = options?.group ?? '';
```

Reply:
So... if the function `log()` is called without `group` parameter, its value will be `undefined`. If that is `undefined`, we do not want to call `core.startGroup()` and `core.endGroup()` otherwise we will have non-deterministic name for the group. Since it is a user-land value, we don't really have a way to meaningfully guess it. So, if we give it a default value of an empty string, what would that do?

Reply:
ok so deliberate, that's fine 💯